### ISSUE TITLE
onnx2coreml conversion issue

### ISSUE THREAD

**Issue by dzyjjpy** [Ma-Dan/Yolact-CoreML#1]
Hi @Ma-Dan , a few conversion question confused me. pls share your experience.
**1.Run yolcat code with ONNX and CoreML converter to convert to ONNX model (WITHOUT priors layer).**
Q: I converted onnx model successfully follow your code in github, remove dict and priors. Test okay
2.Use onnx-simplifier to simplify ONNX model.
Q: is that neccessary?  I just want to run the model on ios device. And do not compress the model now.
3.Disable Upsample layer in onnx-coreml package to make Upsample custom layer in CoreML.
Q:  Do you mean add parameter in function:
```
coreml_model = convert(model_proto,
					   None,
                       image_input_names=['image'],
					   add_custom_layers=True,   ## !!! do you mean add this option???
					   # minimum_ios_deployment_target='13'   )
```

4.Run ONNXToCoreML converter to conver ONNX mode to CoreML model, input should be MLMultiArray (3x550x550) and we need to do input normalization in our application code.
So you use the coreml model converted by this branch???
https://github.com/Ma-Dan/yolact/tree/coreml

pls help to ensure the branch can convert yolact coreml model successfully or not.
Soooooooo  confused right now, and it takes me long time to debug it. If the branch can't get the coreml, pls let me know, and I won't spend my time on it. Thanks

**Comment by dzyjjpy** [comment]
https://stackoverflow.com/questions/62318038/xrunc-coreml-model-error-for-yolact-onnx-with-no-priors-layer-and-softmax-layer

@Ma-Dan could you pls have a look about the onnx2coreml issue?  

**I have convert the YOLACT pytorch model to onnx with no softmax and priors layers.** And I try to convert onnx to coreml afterward. **The terminal shows it's done with no mistake.** It also display model compilation done, and no error as below.

210/211: Converting Node Type Concat
211/211: Converting Node Type Concat
Translation to CoreML spec completed. Now compiling the CoreML model.
Model Compilation done.

But as I compile coreml model on macos, error shows:

**xcrun coremlc compile yolact_test_nosoftmax_simplify.mlmodel**
**coremlc: Error: compiler error:  Espresso exception:
Invalid blob shape  generic_elementwise_kernel: cannot broadcast [18, 18, 128, 1, 399] and [35, 35, 128, 1, 399]**

**Comment by muratatakashi** [comment]
I had the same issue.

3.Disable Upsample layer in onnx-coreml package to make Upsample custom layer in CoreML.

I understood this to be commenting out the Upsample from _ONNX_NODE_REGISTRY in _operators.py and _operators_nd.py in onnx-coreml.
Now I can finally use the converted mlmodel on iOS.

It's not a good way to do it, but I couldn't find a way to specify it with the convert function options etc.

I hope this will be helpful to you.

**Comment by muratatakashi** [comment]
https://github.com/Ma-Dan/yolact/tree/coreml

The above onnx_to_coreml.py could also be modified as follows.
This would eliminate the need to edit the onnx-coreml source file.

`import sys`
`from onnx import onnx_pb`
`from onnx_coreml import convert`
`from onnx_coreml._operators import _ONNX_NODE_REGISTRY`
`from onnx_coreml._operators_nd import _ONNX_NODE_REGISTRY_ND`

`# Disable Upsample Layer`
`del(_ONNX_NODE_REGISTRY['Upsample'])`
`del(_ONNX_NODE_REGISTRY_ND['Upsample'])`
`... convert operations`
